server/src/main/java/io/pravega/schemaregistry/storage/impl/SchemaChunks.java

```diff
@@ -0,0 +1,91 @@
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Utility class for chunking a schema info into multiple chunks if the serialized schema info may exceed the limit of 
+ * KVT key+value size limit of 1 Mb. 
+ * We set the chunking limit as 900 Kb so that we have about 100 Kb for additional schema info fields and serialization overheads. 
+ * The schema binary is chunked into chunks of 900 Kb each. 
+ * So if there is a schema of 8 Mb in size, the number of chunks will be 10 chunks overall.
+ * The first chunk is always included in the schema info object while remaining chunks are presented as list of byte array segments.
+ */
+@Data
+public class SchemaChunks {
+    private static final int MAX_CHUNK_SIZE = 921600; // setting max chunk size to 900 kb as KVT allows for key + value to be 1mb
+
+    private final SchemaInfo schemaInfo;
+    private final List<ByteArraySegment> chunks;
+
+    private SchemaChunks(SchemaInfo schemaInfo, List<ByteArraySegment> chunks) {
+        this.schemaInfo = schemaInfo;
+        this.chunks = chunks;
+    }
+
+    /**
+     * Chunks a schema binary from {@link SchemaInfo#getSchemaData()} into multiple smaller chunks by dividing the array
+     * into smaller arrays of maximum 900 kb size. This returns a schema chunk object which can be assembled back into
+     * original schema info object. 
+     * The schema chunk contains a schema info, which has all the fields and a chunked schema data that has the first chunk.
+     * The remaining chunks are available as list of byte array segments.  
+     * @param schemaInfo Schema info to chunk
+     * @return SchemaChunks object
+     */
+    public static SchemaChunks chunk(SchemaInfo schemaInfo) {
+        int numberOfChunks = schemaInfo.getSchemaData().remaining() % MAX_CHUNK_SIZE == 0 ?
+                schemaInfo.getSchemaData().remaining() / MAX_CHUNK_SIZE - 1 :
+                schemaInfo.getSchemaData().remaining() / MAX_CHUNK_SIZE;
+        int firstChunkSize = numberOfChunks > 0 ? MAX_CHUNK_SIZE : schemaInfo.getSchemaData().remaining();
+
+        ByteBuffer firstChunk = ByteBuffer.wrap(schemaInfo.getSchemaData().array(), schemaInfo.getSchemaData().position(), firstChunkSize);
+        SchemaInfo chunkedSchemaInfo = new SchemaInfo(schemaInfo.getType(), schemaInfo.getSerializationFormat(),
+                firstChunk, schemaInfo.getProperties());
+        List<ByteArraySegment> chunks = new ArrayList<>();
+
+        // break schema binary into smaller chunks.
+        for (int i = 1; i <= numberOfChunks; i++) {
+            int offset = i * MAX_CHUNK_SIZE;
+            int length = schemaInfo.getSchemaData().remaining() > offset + MAX_CHUNK_SIZE ? MAX_CHUNK_SIZE :
+                    schemaInfo.getSchemaData().remaining() - offset;
+            ByteArraySegment chunk = new ByteArraySegment(schemaInfo.getSchemaData().array(), offset, length);
+            chunks.add(chunk);
+        }
+        return new SchemaChunks(chunkedSchemaInfo, chunks);
```

Comment:
The class design looks odd to me. A `SchemaInfo` object has (is built of) chunks, so `SchemaInfo` should comprise of chunks, not the other way around. 

Can we have this functionality moved inside of `SchemaInfo`? If you do that, `chunk()` would become a method of `SchemaInfo`, and you could have `SchemaInfo.fromChunks(SchemaChunks)` return you a an SchemaInfo.